geonode/base/api/views.py

```diff
@@ -225,11 +225,15 @@ class HierarchicalKeywordViewSet(WithDynamicViewSetMixin, ListModelMixin, Retrie
     """
     API endpoint that lists hierarchical keywords.
     """
+    def get_queryset(self):
+        resource_keywords = HierarchicalKeyword.resource_keywords_tree(self.request.user)
```

Comment:
Why are we going to pass the `request.user` here? Shouldn't be sufficient to filter out the objects without an `href`?

Reply:
Because the [function](https://github.com/GeoNode/geonode/blob/master/geonode/base/models.py#L344) checks resources a user is permitted to access, and it returns a list of a dict with this [structure](https://github.com/GeoNode/geonode/blob/master/geonode/base/models.py#L384)